frontend/src/components/Search/SearchForm/SearchForm.tsx

```diff
@@ -645,6 +647,20 @@ export const Search: React.FC<{
 		comboboxStore.setState('moves', 0)
 	}
 
+	const handleHistorySelction = (
+		query: string,
+		queryParts: SearchExpression[],
+	) => {
+		const newQuery = stringifySearchQuery(queryParts)
+		startTransition(() => {
+			submitQuery(newQuery)
+			handleSetCursorIndex()
+			comboboxStore.setOpen(false)
+		})
+		comboboxStore.setActiveId(null)
+		comboboxStore.setState('moves', 0)
```

Comment:
sorry for all the back-and-forth, this should be the last feedback point
https://www.loom.com/share/f3d046d53f764dfabf6080f8e4f6cf78